src/ecs/resources/input.rs

```diff
@@ -62,45 +80,63 @@ impl<'a> Iterator for PressedMouseButtons<'a> {
 /// ```
 #[derive(Default)]
 pub struct InputHandler {
-    pressed_keys: HashMap<VirtualKeyCode, KeyQueryState>,
-    pressed_mouse_buttons: HashMap<MouseButton, KeyQueryState>,
+    pressed_keys: Vec<VirtualKeyCode>,
```

Comment:
I had similar thoughts, I initially wanted to store a fixed size array of bools as part of the struct, where each `VirtualKeyCode` (or MouseButton) `as usize` is the index into the array.  This reduces the amount of times we have to follow pointers and do allocations, but determining what that fixed size should be is a bit hard as there isn't an easy way to get a count of all the variants of a `VirtualKeyCode`.  I could just hard code a value but this makes for some brittle code if new codes are ever added.  Additionally every frame the current input state needs to be copied to the previous input state as well, so if the input state structures are larger this gets more expensive.  There are also limitations on how large this Vec<VirtualKeyCode> can get, most of the time it will have < 5 elements, so iterating it to determine key state isn't that expensive, especially when compared to the cost of copying the entire keyboard structure every frame.